### PR TITLE
Should be possible to change and save money back to default currency

### PR DESCRIPTION
**Pull request by benjaoming**
#221

### CODE REVIEW

**Comment by benjaoming** [comment]
The following test breaks when changing the behavior: 

```
    @pytest.mark.parametrize(
        'kwargs, currency',
        (
            ({'money_currency': 'PLN'}, 'PLN'),
            ({'money': Money(0, 'EUR')}, 'EUR')
        )
    )
    def test_get_or_create_respects_currency(self, kwargs, currency):
        instance, created = ModelWithVanillaMoneyField.objects.get_or_create(**kwargs)
>       assert str(instance.money.currency) == currency, 'currency should be taken into account in get_or_create'
E       AssertionError: currency should be taken into account in get_or_create
E       assert 'XYZ' == 'PLN'
E         - XYZ
E         + PLN
```

**Comment by benjaoming** [comment]
The explanation is already in an older comment:

```
         # we have to determine whether to replace the currency.
         # i.e. if we do the following:
         # .objects.get_or_create(money_currency='EUR')
         # then the currency is already set up, before this code hits
         # __set__ of MoneyField. This is because the currency field
         # has less creation counter than money field.
```

**Comment by benjaoming** [comment]
Any advice? Ideas?

**Comment by Stranger6667** [comment]
@benjaoming 
I'll take a look tomorrow :)

**Comment by Stranger6667** [comment]
It seems like our builds are broken, I'll fix it tomorrow and rerun all PRs

**Comment by Stranger6667** [comment]
For me it seems perfectly fine! Probably we could just set currency like this:

```python
    def prepare_value(self, obj, value):
        validate_money_value(value)
        currency = get_currency(value)
        if currency:
            setattr(obj, self.currency_field_name, currency)
        return self.field.to_python(value)
```

`get_or_create` fails only if there is no info about `money` field:

Good:
```python
{'money': Money(0, 'EUR')}
```

Also good:

```python
{'money': 10, 'money_currency': 'PLN'}
```

Bad:
```
{'money_currency': 'PLN'}
```

Now we manipulate with currency if currency is absent - we're setting default value for it.
Probably we should consider opposite situation, when we have only currency and no amount. We can just take default value for amount.
Something like this (`models.managers.py`):

```python
from .fields import MoneyField, CurrencyField

...

def _expand_money_kwargs(model, args=(), kwargs=None, exclusions=()):
    """
    Augments kwargs so that they contain _currency lookups.
    """
    involved_fields = [_get_clean_name(name) for name in kwargs]
    for name, value in list(kwargs.items()):
        if name in exclusions:
            continue
        if isinstance(value, Money):
            clean_name = _get_clean_name(name)
            kwargs[name] = value.amount
            kwargs[get_currency_field_name(clean_name)] = smart_unicode(value.currency)
        else:
            field = _get_field(model, name)
            if isinstance(field, MoneyField):
                if isinstance(value, (BaseExpression, F)):
                    clean_name = _get_clean_name(name)
                    if not isinstance(value, F):
                        value = prepare_expression(value)
                    kwargs[get_currency_field_name(clean_name)] = F(get_currency_field_name(value.name))
                if is_in_lookup(name, value):
                    args += (_convert_in_lookup(model, name, value), )
                    del kwargs[name]
            elif isinstance(field, CurrencyField):
                money_field_name = name[:-9]  # Remove '_currency'
                if money_field_name not in involved_fields:
                    money_field = _get_field(model, money_field_name)
                    kwargs[money_field_name] = money_field.default.amount

    return args, kwargs
```

What do you think? All tests pass with these changes. I tried Django 1.4, 1.10.

**Comment by benjaoming** [comment]
@Stranger6667 thanks, seems to make sense -- so with the changes, it makes it possible to do all this?

    MyModel.objects.filter(money_currency="USD")

...or was this functional even before the change?

**Comment by Stranger6667** [comment]
It was functional :)

**Comment by benjaoming** [comment]
Hmm, not sure :) would you like to write a sentence for the changelog? :)

**Comment by Stranger6667** [comment]
Ah, I thought, that you mean some extra line about `get_or_create` behaviour. 
Sure, I'll add some about this issue :) Thank you

**Comment by benjaoming** [comment]
Oh snap, I thought I had already written release notes for the fix... I did that only in the other PR yesterday.. I'll add something here...

**Comment by benjaoming** [comment]
Rebased and added release note

File: CHANGES.rst [hunk 1]
```diff
@@ -219,4 +220,7 @@ Changes in 0.3
 .. _tsouvarev: https://github.com/tsouvarev
 .. _w00kie: https://github.com/w00kie
 .. _willhcr: https://github.com/willhcr
+<<<<<<< e6efad2071e1281609732cc5c9d5c1cbf5c17c18
```
Comment (Stranger6667) [inline CHANGES.rst]: Its probably should not be there :)

Comment (Stranger6667) [inline CHANGES.rst]: I'll fix it

Comment (benjaoming) [inline CHANGES.rst]: No indeed :)

Comment (benjaoming) [inline CHANGES.rst]: Oh I just force-pushed the fix, wonder who made it first

Comment (benjaoming) [inline CHANGES.rst]: You did ;)

**Pull request merged**